### DISCUSSION TITLE
Watchtower fails to kill containers

### DISCUSSION THREAD

**Discussion by socram70** [containrrr/watchtower#1887] [Q&A] [answered]
Hi all,

I keep getting the following messages when trying to update my containers:

```
time="2023-12-18T19:07:54Z"` level=error msg="Error response from daemon: Cannot kill container: 1b1d48562c9d87bca921f2f9b22cdc634f4c7d8e26de1818bd4bf7c31d915458: Cannot kill container 1b1d48562c9d87bca921f2f9b22cdc634f4c7d8e26de1818bd4bf7c31d915458: unknown error after kill: runc did not terminate successfully: exit status 1: unable to signal init: permission denied\n: unknown"

time="2023-12-18T19:07:54Z" level=info msg="Session done" Failed=3 Scanned=20 Updated=0 notify=no

time="2023-12-18T19:08:09Z" level=info msg="Found new nextcloud:fpm-alpine image (ce840e816be5)"
```

Does anyone have an idea how this could be solved?

I already tried to use watchtower via the stack function in portainer and also via a separate docker-compose file (outside of portainer) - none helped.

Docker version 24.0.5, build 24.0.5-0ubuntu1
Linux version 6.5.0-14-generic (buildd@lcy02-amd64-031) (x86_64-linux-gnu-gcc-13 (Ubuntu 13.2.0-4ubuntu3) 13.2.0, GNU ld (GNU Binutils for Ubuntu) 2.41) #14-Ubuntu SMP PREEMPT_DYNAMIC Tue Nov 14 14:59:49 UTC 2023

**Comment by piksel** [comment]
Well, watchtower asks docker to stop a container and it just fails to do so. I don't think the way you setup watchtower matters in this situation. 
Perhaps you have some additional security rules that prevents containers from stopping each other? Where are you running your containers? is it some kind of vendor-customized machine perhaps?

**Comment by socram70** [comment, marked as answer]
Thank you very much @piksel for your time. I thought the same and therefore figured out that apparmor in the newest Ubuntu release might cause these problems. If anybody else ever stumbles upon this issue, this is how I solved it:

sudo aa-remove-unknown
sudo apt purge --auto-remove apparmor
sudo service docker restart

The first command might be sufficient, so I wouldn't bother uninstalling apparmor, since this also uninstalls snapd.

**Reply by ayamdobhal** [reply]
i can't quite figure what firefox is supposed to be doing here?

**Reply by NeerajGodiyal** [reply]
> i can't quite figure what firefox is supposed to be doing here?

same question

**Reply by imsk17** [reply]
+1

**Reply by luisdralves** [reply]
Ubuntu installs Firefox as a snap. You can add the Mozilla PPA as an alternative.

**Reply by ayamdobhal** [reply]
> Ubuntu installs Firefox as a snap. You can add the Mozilla PPA as an alternative.

Still makes no sense?

**Reply by ayamdobhal** [reply]
okay nvm saw the edit
